src/main/java/Duke.java

```diff
@@ -1,10 +1,272 @@
+     */
+    public static void printTaskList() {
+        System.out.println("| Task List |");
+        if (inputCount == 0) {
+            System.out.println(ERROR_EMPTY_LIST);
+        } else {
+            for (int i = 0; i < inputCount; i++) {
+                System.out.println((i + 1) + ". " + tasks[i].toString());
+            }
+        }
+    }
+
+    /**
+     * Modifies the task identified by the input task number and print out a confirmation message.
+     *
+     * @param shouldMarkTask Determines whether to set task to done or not done
+     * @param taskNoString   The input task number that identifies the task in "tasks"
+     */
+    public static void markTask(boolean shouldMarkTask, String taskNoString) {
+        // Index in Array
+        int taskNo;
+        try {
+            taskNo = Integer.parseInt(taskNoString) - 1;
+        } catch (NumberFormatException exception) {
+            System.out.println(ERROR_INVALID_TASK_NUMBER);
+            System.out.println("Please enter a value between 1 and " + inputCount);
+            return;
+        }
+
+        if (taskNo < 0 || taskNo >= inputCount) {
+            System.out.println(ERROR_INVALID_TASK_NUMBER);
+            if (inputCount == 0) {
+                System.out.println(ERROR_EMPTY_LIST);
+            } else {
+                System.out.println("Please enter a value between 1 and " + inputCount);
+            }
+        } else {
+            // Check if already marked / unmarked
+            if (tasks[taskNo].isDone() == shouldMarkTask) {
+                System.out.println("Your task has already been marked as " + (shouldMarkTask ? "done!" : "not done!"));
+            } else {
+                tasks[taskNo].setDone(shouldMarkTask);
+                System.out.println("Your task is now marked as " + (shouldMarkTask ? "done!" : "not done!"));
+            }
+            System.out.println("> " + (taskNo + 1) + ". " + tasks[taskNo].toString());
+        }
+    }
+
+    /**
+     * Reads in input from standard input and processes the input to determine
+     * what operations to carry out by calling other helper methods.
+     */
+    public static void processInput() {
+        Scanner in = new Scanner(System.in);
+        System.out.print(INPUT_PREPEND);
+        String input = in.nextLine();
+        String[] inputs = input.split(REGEX_PATTERN_WHITESPACES, 2);
+
+        boolean isCommandOnly;
+        while (inputs.length > 0) {
+            printDivider();
+            switch (inputs[0]) {
+            case "": // EMPTY INPUT
+                System.out.println(ERROR_EMPTY_INPUT);
+                break;
+            case "?": // DISPLAY HELP MENU, FALLTHROUGH, IDENTICAL COMMAND
+            case "help":
+                System.out.println(HELP_MESSAGE);
+                break;
+            case "list": // LIST OUT TASK LIST
+                printTaskList();
+                break;
+            case "mark": // MARK TASK AS DONE
```

Comment:
Maybe the comments would be good to be on the line above as compared to being inline